packages/dev/core/src/Engines/WebGPU/Extensions/engine.computeShader.ts

```diff
@@ -10,11 +10,21 @@ import { WebGPUComputeContext } from "../webgpuComputeContext";
 import { WebGPUComputePipelineContext } from "../webgpuComputePipelineContext";
 import * as WebGPUConstants from "../webgpuConstants";
 import type { WebGPUPerfCounter } from "../webgpuPerfCounter";
+import type { DataBuffer } from "../../../Buffers/dataBuffer";
 
 declare module "../../webgpuEngine" {
     export interface WebGPUEngine {
         /** @internal */
         _createComputePipelineStageDescriptor(computeShader: string, defines: Nullable<string>, entryPoint: string): GPUProgrammableStage;
+        /** @internal */
+        _computeDispatch(
+            effect: ComputeEffect,
+            context: IComputeContext,
+            bindings: ComputeBindingList,
+            options: { x: number; y: number; z: number } | { buffer: DataBuffer; offset: number },

```

Comment:
Same comment as above, we should not create new objects each time `computeDispatch` or `computeDispatchIndirect` are called.